webui.py

```diff
@@ -621,8 +624,31 @@ def save_sample(image, sample_path_i, filename, jpg_sample, prompts, seeds, widt
             #info_dict["init_mask"] = init_mask
             info_dict["denoising_strength"] = denoising_strength
             info_dict["resize_mode"] = resize_mode
-        with open(f"{filename_i}.yaml", "w", encoding="utf8") as f:
-            yaml.dump(info_dict, f, allow_unicode=True, width=10000)
+        if write_info_files:
+            with open(f"{filename_i}.yaml", "w", encoding="utf8") as f:
+                yaml.dump(info_dict, f, allow_unicode=True, width=10000)
+
+        if write_info_to_one_file:
+            ignore_list = ["prompt", "target", "toggles", "ddim_eta", "batch_size"]
+            rename_dict = {"ddim_steps": "steps", "n_iter": "number", "sampler_name": "sampler"} #changes the name of parameters to match with dynamic parameters
+            sample_log_path = os.path.join(sample_path_i, "log.yaml")
+            log_dump = info_dict.get("prompt") # making sure the first item that is listed in the txt is the prompt text
+            for key, value in info_dict.items():
+                if key in ignore_list:
+                    continue
+                found_key = rename_dict.get(key)
+
+                if key == "cfg_scale": #adds zeros to to cfg_scale necessary for dynamic params
+                    value = str(value).zfill(2)
+
+                if found_key:
+                    key = found_key
+                log_dump += f" {key} {value}"

```

Comment:
This will be difficult to read.
I recommend using different separators between key and value and two pairs of key and value.